cpp/oneapi/dal/backend/primitives/select.cpp

```diff
@@ -0,0 +1,249 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <algorithm>
+#include <utility>
+#include <exception>
+#include <limits>
+#include <list>
+
+#ifdef ONEAPI_DAL_DATA_PARALLEL
+    #include <CL/sycl.hpp>
+#endif
```

Comment:
This one is already included by `oneapi/dal/backend/primitives/common.hpp` or `array.hpp`